Licenses for maps!

KingOfSquares opened the pull request:
R.I.P Sherman, fixes #496 

Made under new: "OCC plays IntelliJ" show

Any other licenses to add?

Commit:
Licenses for maps!

Signed-off-by: SimonProMac <simonmr60@gmail.com>
Signed-off-by: KingSimon <simonmorland@gmail.com>

KingOfSquares commented:
Closes #496 btw

Commit:
Add the option for a custom license
"Hey, maybe turn off your IntelliJ formatter you dingus"

Signed-off-by: KingSimon <simonmorland@gmail.com>

Electroid commented on core/src/main/java/tc/oc/pgm/map/MapLicense.java:
```diff
@@ -0,0 +1,29 @@
+  CC_BY_NC("Attribution-NonCommercial 4.0 International"),
+  CC_BY_NC_ND("Attribution-NonCommercial-NoDerivatives 4.0 International"),
+  CC_BY_NC_SA("Attribution-NonCommercial-ShareAlike 4.0 International"),
+  CUSTOM(""),
```
While I see the desire for a custom license, I'm not convinced we should allow this. This problem is similar to custom map rules: they cause confusion. I don't want map makers defining their own licenses, that's murky water. Creative Commons is used by basically everyone, and it has good legal standing.

KingOfSquares replied:
Yes! Was wondering about this as well. I am going to remove it, as the implementation is kinda weird as well.

Electroid commented on core/src/main/java/tc/oc/pgm/map/MapLicense.java:
```diff
@@ -0,0 +1,29 @@
+package tc.oc.pgm.map;
+
+public enum MapLicense {
```
Isn't there a public domain license from Creative Commons?

Electroid replied:
Is there anyway we can use their translations? Example: https://github.com/creativecommons/cc.i18n/blob/master/cc/i18n/po/fr_CA/cc_org.po#L639

KingOfSquares replied:
So a way to fetch those at runtime(assuming we don't want to bundle these translations with the rest)?

Electroid commented on core/src/main/java/tc/oc/pgm/command/MapCommand.java:
```diff
@@ -208,6 +208,12 @@ public void map(Audience audience, CommandSender sender, @Text MapInfo map) {
+    audience.sendMessage(
+        TextComponent.builder()
+            .append(mapInfoLabel("map.info.license"))
+            .append(TextComponent.of(map.getLicense().getFullName(), TextColor.GOLD))
```
Can we get a hover click to view the license? Also can we make the license more human-readable / understandable? Tags like "Commercial" or something?

Assume the user doesn't know much about CC.

KingOfSquares replied:
Yes, great ideas!

KingOfSquares replied:
I tried changing them a little, let me know if this is what you were thinking. I think keeping the words as they are in the license is logical because if you google "Attribute license" the CC license will appear.

BGMP commented:
@Shrman007658 would be proud

Commit:
Remove custom licenses++
- make names more readable
- add links to each license to read more about them

Signed-off-by: KingSimon <simonmorland@gmail.com>

KingOfSquares commented:
I can't see myself doing this In while. Will re-open if anything happenes with this project
